### PR TITLE
[WIP] Upgrade prerelease dependencies to rc2-23907.

### PR DESCRIPTION
**Pull request by dagood**
Update to latest packages based on discussion in https://github.com/dotnet/buildtools/pull/510/.

I'm getting some restore errors with 23907:

	Errors in E:\git\buildtools\src\Microsoft.DotNet.Build.Tasks.Packaging\src.Desktop\project.json
	  System.Runtime.InteropServices.RuntimeInformation 4.0.0-rc2-23907 provides a compile-time reference assembly for System.Runtime.InteropServices.RuntimeInformation on .NETFramework,Version=v4.5, but there is no run-time assembly compatible with win.
	  Some packages are not compatible with .NETFramework,Version=v4.5 (win).

Looking in the package, the runtime.json that I think previously let the restore find the runtime package is no longer there. Not sure if this is a bug or if there's some new way to get to runtimes that I'm not aware of. I can work around this by depending on the win runtime package manually, but that's nasty.

	Errors in E:\git\buildtools\src\Microsoft.DotNet.Build.Tasks\project.json
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.10-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.10-x64).
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with ubuntu.14.04-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (ubuntu.14.04-x64).
	Errors in E:\git\buildtools\src\Microsoft.DotNet.Build.Tasks\PackageFiles\tool-runtime\project.json
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with centos.7-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (centos.7-x64).
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with debian.8.2-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (debian.8.2-x64).
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.10-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.10-x64).
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with rhel.7-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (rhel.7-x64).
	  Microsoft.Win32.Registry 4.0.0-rc2-23907 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with ubuntu.14.04-x64.
	  Some packages are not compatible with DNXCore,Version=v5.0 (ubuntu.14.04-x64).

Microsoft.Win32.Registry 23816 had `lib/dnxcore50/Microsoft.Win32.Registry.dll`, but 23907 only has `runtimes\win\lib\netstandard1.3\Microsoft.Win32.Registry.dll`. Since this dependency is brought in by `Microsoft.Build.Tasks.Core` and `Microsoft.Build.Utilities.Core` I'm not sure how to fix this... normally for a package that only exists on one platform I would fork the project.json, but it seems like we would hit this same problem when trying to restore tool-runtime on an actual xplat machine. (Because we do want these `Microsoft.Build...` packages on every platform, right?)

/cc @stephentoub @ericstj @weshaggard

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src.Desktop/project.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "Newtonsoft.Json": "7.0.1",
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src.Desktop/project.json]: Does updating to the latest Platforms package in this project fix the error you are seeing?

**Comment by ericstj** [comment]
This is happening because we stopped putting runtime.json in the reference packages.  Instead, the sole runtime.json should be coming from the lineup.  For RuntimeInformation we need to make sure that project includes the lineup (MS.NETCore.Platforms).  I also needed to update this project to net46 in order to bring in the lineup.  We should get that fixed in the packages.

For Registry that is a valid compat error.  To avoid it we can suppress the compile reference for registry:
    "Microsoft.Win32.Registry": { 
      "version": "4.0.0-rc2-23907",
      "exclude": "Compile"
    }

I did this in build tools and it resolved the restore error, but I noticed that package validation cannot handle this syntax (CLI does).

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src.Desktop/project.json (hunk 1 above)
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src.Desktop/project.json]: If I change this *and* change the framework to `net46` it works, but with `net45` it doesn't get the `Microsoft.NETCore.Targets.NETFramework` dependency. Is 4.6 vs. 4.5 a problem?

**Comment by dagood** [comment]
Ok, thanks for the help, I fixed this up.

**Comment by ericstj** [comment]
Can you test consumption of this in CoreFx and TFS before merging?

**Comment by dagood** [comment]
Sure.

**Comment by dagood** [comment]
I was able to build from FxCore in TFS, but apparently my enlistment's full build isn't working.

Build.cmd on CoreFX seemed to go ok, but I hit what appears to be the test failure discussed here: https://github.com/dotnet/corefx/pull/6545#issuecomment-190706407. I'm also seeing a bunch of packaging errors like this now:

	  System.IO -> C:\git\corefx\bin/pkg/specs/runtime.aot.System.IO.nuspec
	C:\git\corefx\Tools\Packaging.targets(842,5): error : SupportedFramework DNXCore,Version=v5.0 was specified but is not part of the Framework list to use for validation. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : System.IO.FileSystem.Watcher should not be supported on .NETStandardApp,Version=v1.5/win7-x86 but has both compile and runtime assets. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : System.IO.FileSystem.Watcher should not be supported on .NETStandardApp,Version=v1.5/win7-x64 but has both compile and runtime assets. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : System.IO.FileSystem.Watcher should not be supported on .NETStandardApp,Version=v1.5/osx.10.11-x64 but has both compile and runtime assets. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : System.IO.FileSystem.Watcher should not be supported on .NETStandardApp,Version=v1.5/centos.7.1-x64 but has both compile and runtime assets. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : System.IO.FileSystem.Watcher should not be supported on .NETStandardApp,Version=v1.5/ubuntu.14.04-x64 but has both compile and runtime assets. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : System.IO.FileSystem.Watcher should not be supported on .NETStandardApp,Version=v1.5/linuxmint.17-x64 but has both compile and runtime assets. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : Framework Xamarin.TVOS,Version=v1.0 should support System.IO.FileSystem.Watcher inbox but contained a reference assemblies: System.IO.FileSystem.Watcher/ref/netstandard1.3/System.IO.FileSystem.Watcher.dll.  You may need to add <InboxOnTargetFramework Include="xamarintvos10" /> to your project. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : Framework Xamarin.TVOS,Version=v1.0 should support System.IO.FileSystem.Watcher inbox but was missing a placeholder for run-time.  You may need to add <InboxOnTargetFramework Include="xamarintvos10" /> to your project. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : Framework Xamarin.WatchOS,Version=v1.0 should support System.IO.FileSystem.Watcher inbox but contained a reference assemblies: System.IO.FileSystem.Watcher/ref/netstandard1.3/System.IO.FileSystem.Watcher.dll.  You may need to add <InboxOnTargetFramework Include="xamarinwatchos10" /> to your project. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]
	C:\git\corefx\Tools\Packaging.targets(842,5): error : Framework Xamarin.WatchOS,Version=v1.0 should support System.IO.FileSystem.Watcher inbox but was missing a placeholder for run-time.  You may need to add <InboxOnTargetFramework Include="xamarinwatchos10" /> to your project. [C:\git\corefx\src\System.IO.FileSystem.Watcher\pkg\System.IO.FileSystem.Watcher.pkgproj]

This is a buildtools upgrade from 00177 to 00189--it seems like a lot could have changed through then. Any ideas @ericstj?

/cc @stephentoub @weshaggard

**Comment by ericstj** [comment]
You need to pull latest CoreFx, it should be on a much newer buildtools.

**Comment by ericstj** [comment]
See https://github.com/dotnet/corefx/blob/master/BuildToolsVersion.txt : 1.0.25-prerelease-00190

**Comment by dagood** [comment]
Ah, yep, I mixed up what I had cloned and was actually using my fork's 14-day-old master. (Now I've deleted that so it can't happen again.) Rerunning the build, will merge if successful.

**Comment by dagood** [comment]
Building CoreFX, all I get now is the known error https://github.com/dotnet/corefx/pull/6747. Merging!

**Comment by ericstj** [comment]
Awesome.  Thank you Davis.

**Comment by stephentoub** [comment]
Thanks, @dagood!

**Pull request merged**